tools/crm_error.c

```diff
@@ -78,11 +79,97 @@ get_strings(int rc, const char **name, const char **str)
     }
 }
 
+PCMK__OUTPUT_ARGS("result-code", "int", "char *", "char *");
+static int
+result_code_default(pcmk__output_t *out, va_list args)
+{
+    int code = va_arg(args, int);
+    char *name = va_arg(args, char *);
+    char *desc = va_arg(args, char *);
+
+    char *min_rc_str = NULL;
+    int code_width = 0;
+
+    if (out->is_quiet(out)) {
+        /* If out->is_quiet(), we don't print the code -- so if name and desc
+         * are NULL, there's nothing to print. It makes no sense to call the
+         * formatter that way, but it's probably not worth an assert.
+        */
+        if ((name != NULL) && (desc != NULL)) {
+            pcmk__formatted_printf(out, "%s - %s\n", name, desc);
+        } else if ((name != NULL) || (desc != NULL)) {
+            pcmk__formatted_printf(out, "%s\n", ((name != NULL)? name : desc));
+        }
+        return pcmk_rc_ok;
+    }
```

Comment:
You could just pass an extra bool argument to result_code_default to control whether to print this or not.  Various other format functions take flags for things like this.

Reply:
That's valid too. My thinking is that the bool would be ignored in the non-default (i.e., non-text) context, and that the `quiet` flag also conveys that it enables a compact form of the output.